### ISSUE TITLE
ChainerX BatchNorm unstable for FP16

### ISSUE THREAD

**Issue by hvy** [chainer/chainer#6609]
The `BatchNorm` is quite unstable with FP16, both native and CUDA and should be debugged. The backward tests in Python for instance fail unless relaxed with high tolerances (higher than Chainer's  `F.batch_normalization` tests).

**Comment by hvy** [comment]
https://github.com/chainer/chainer/blob/master/tests/chainerx_tests/unit_tests/routines_tests/test_normalization.py#L106-L110

**Comment by emcastillo** [comment]
https://discuss.pytorch.org/t/training-with-half-precision/11815

Seems like this is a global issue due to the numerical format itself
